src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1214,4 +1233,45 @@ protected Engine newEngine() {
     public boolean allowsPrimaryPromotion() {
         return true;
     }
+
+    // pkg private for testing
+    void persistMetadata(ShardRouting newRouting, ShardStateMetaData oldShardStateMetadata) {
```

Comment:
the reason is based on the oldShard so If pass it down I basically just move things around which is odd but I can do it. We will will have to stick to the oldShardStateMetadata so I left is was it is. Thoughts?